### DISCUSSION TITLE
How make to variable menu, like ｢for each i in｣

### DISCUSSION THREAD

**Discussion by NOname-awa** [TurboWarp/extensions#1114] [Q&A]
I want to make a block about ｢for each｣.
At first, I want to use XML block, but I found out that the variable menu is not menu, it seems like this is a type

`
      "args0": [
        {
          "type": "field_variable",
          "name": "VARIABLE"
        }
`

Then I want to write the same variable menu, but I don't know how make to it (variable menu).

**Comment by Skyhigh173** [comment]
~~steal~~ reference some code from JSON extension

**Reply by NOname-awa** [reply]
but「JSON」extension don’t have var menu

**Reply by Skyhigh173** [reply]
> but「JSON」extension don’t have var menu

there's a list menu. you can modify it by changing the filter type from `"list"` to `""` (empty string).

**Reply by NOname-awa** [reply]
but「.id」and「.name」…
in「for」block, var menu return is a object (I'm not sure)
this is source code for「for」block:
```
 ​    ​forEach​ ​(​args​,​ ​util​)​ ​{ 
 ​        ​const​ ​variable​ ​=​ ​util​.​target​.​lookupOrCreateVariable​( 
 ​            ​args​.​VARIABLE​.​id​,​ ​args​.​VARIABLE​.​name​)​; 
  
 ​        ​if​ ​(​typeof​ ​util​.​stackFrame​.​index​ ​===​ ​'undefined'​)​ ​{ 
 ​            ​util​.​stackFrame​.​index​ ​=​ ​0​; 
 ​        ​} 
  
 ​        ​if​ ​(​util​.​stackFrame​.​index​ ​<​ ​Number​(​args​.​VALUE​)​)​ ​{ 
 ​            ​util​.​stackFrame​.​index​++​; 
 ​            ​variable​.​value​ ​=​ ​util​.​stackFrame​.​index​; 
 ​            ​util​.​startBranch​(​1​,​ ​true​)​; 
 ​        ​} 
 ​    ​}
```

**Reply by CubesterYT** [reply]
@NOname-awa, there is a variable menu in the tween extension.

**Reply by yuri-kiss** [reply]
you could also use xml ig
<shadow type="data_variablemenu"></shadow>

**Reply by yuri-kiss** [reply]
`<shadow type="data_variablemenu"></shadow>`

**Reply by NOname-awa** [reply]
thanks！

**Comment by NOname-awa** [comment]
> `<shadow type="data_variablemenu"></shadow>`

but it ...
![image](https://github.com/TurboWarp/extensions/assets/126544888/4d5be0f1-797a-41f8-90df-ae76af492d00)

**Reply by yuri-kiss** [reply]
oh i forgot the value wrapper sorry`
    <value name="VARIABLE">
      <shadow type="data_variablemenu"></shadow>
    </value>`

**Reply by NOname-awa** [reply]
> oh i forgot the value wrapper sorry`
>     <value name="VARIABLE">
>       <shadow type="data_variablemenu"></shadow>
>     </value>`

but it is not right too...

**Reply by yuri-kiss** [reply]
worked for me ¯\_(ツ)_/¯

**Reply by NOname-awa** [reply]
> worked for me ¯\_(ツ)_/¯

It might be my fault...
I have currently achieved it by modifying the arg type.
